src/test/unit/types/SdrTest.cpp

```diff
@@ -630,26 +630,30 @@ TEST(SdrTest, TestSaveLoad) {
     ofstream outfile;
     outfile.open(filename);
 
-    // Test zero value
     SDR zero({ 3, 3 });
-    zero.save( outfile );
-
-    // Test dense data
     SDR dense({ 3, 3 });
-    dense.setDense(SDR_dense_t({ 0, 1, 0, 0, 1, 0, 0, 0, 1 }));
-    dense.save( outfile );
-
-    // Test sparse data
     SDR sparse({ 3, 3 });
-    sparse.setSparse(SDR_sparse_t({ 1, 4, 8 }));
-    sparse.save( outfile );
-
-    // Test coordinate data
     SDR coord({ 3, 3 });
-    coord.setCoordinates(SDR_coordinate_t({
-            { 0, 1, 2 },
-            { 1, 1, 2 }}));
-    coord.save( outfile );
+    {
+      cereal::JSONOutputArchive json_out(outfile);
+
+      // Test zero value
+      zero.save_ar( json_out );
+
+      // Test dense data
+      dense.setDense(SDR_dense_t({ 0, 1, 0, 0, 1, 0, 0, 0, 1 }));
+      dense.save_ar( json_out );
+
+      // Test sparse data
+      sparse.setSparse(SDR_sparse_t({ 1, 4, 8 }));
+      sparse.save_ar( json_out );
+
+      // Test coordinate data
+      coord.setCoordinates(SDR_coordinate_t({
+              { 0, 1, 2 },
+              { 1, 1, 2 }}));
+      coord.save_ar( json_out );
+    } // forces Cereal to flush to stream.
```

Comment:
below is 

>  if( false ) {
        // Print the file's contents
        std::stringstream buffer; buffer << infile.rdbuf();
        cout << buffer.str() << "EOF" << endl;
        infile.seekg( 0 ); // rewind to start of file.
    } 

please use the DEBUG functionality you've implemented for cout'ing in gtests

Reply:
> please use the DEBUG functionality you've implemented for cout'ing in gtests

I assume you mean the VERBOSE thing.  Sure I can do that. This is not my code...it was  part of the previous serialization test. So I was making only minimal changes.  (I sort of feel like I am messing around on @ctrl-z-9000-times 's domain.)

Reply:
..ok, so either way you want

Reply:
It's leftover debugging code, you can change it if you want.